kakarot_scripts/deployment/dualvm_token_deployments.py

```diff
@@ -0,0 +1,167 @@
+# %% Imports
+import json
+import logging
+from pathlib import Path
+
+from uvloop import run
+
+from kakarot_scripts.constants import EVM_ADDRESS, NETWORK, RPC_CLIENT, NetworkType
+from kakarot_scripts.utils.kakarot import deploy as deploy_kakarot
+from kakarot_scripts.utils.kakarot import deploy_and_fund_evm_address
+from kakarot_scripts.utils.kakarot import dump_deployments as dump_evm_deployments
+from kakarot_scripts.utils.kakarot import get_contract as get_solidity_contract
+from kakarot_scripts.utils.kakarot import get_deployments as get_evm_deployments
+from kakarot_scripts.utils.starknet import call_contract
+from kakarot_scripts.utils.starknet import deploy as deploy_starknet
+from kakarot_scripts.utils.starknet import get_deployments as get_starknet_deployments
+from kakarot_scripts.utils.starknet import (
+    get_starknet_account,
+    invoke,
+    register_lazy_account,
+    remove_lazy_account,
+)
+from tests.utils.helpers import int_to_string
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+# %%
+async def deploy_dualvm_tokens() -> None:
+    # %% Deploy DualVM Tokens
+
+    # The lazy execution must be done before we check the deployments succeeded, as the l2 contracts
+    # need to be deployed first
+    account = await get_starknet_account()
+
+    # Remove lazy account before we check the deployments succeeded, as the l2 contracts
+    # need to be deployed sequentially.
+    remove_lazy_account(account.address)
```

Comment:
```suggestion
    # Remove lazy execution as the l2 contracts need to be deployed sequentially.
    account = await get_starknet_account()    
    remove_lazy_account(account.address)
```

but I don't understand why this is true actually, why we need to have the Starknet contract deployed to deploy the corresponding dualvm token

Reply:
> but I don't understand why this is true actually, why we need to have the Starknet contract deployed to deploy the corresponding dualvm token

we don't need them to deploy, we need them to to check deployments

if you do

```
        assert await token_contract.name() == token["name"]
```

if the underlying token is not deployed it's going to fail.

we discussed this and you said that it actually didn't matter because the only environment we were deploying starknet tokens in is katana.